keps/sig-apps/3998-job-success-completion-policy/README.md

```diff
@@ -845,13 +855,42 @@ Currently, the indexed job is restricted `.spec.completion!=nil`.
 By relaxing the validation, the indexed job can be declared as succeeded when some indexes succeeded,
 similar to NonIndexed jobs.
 
+### Alternative API Name, "Criteria"
+
+The `criteria` would be matched to express the behavior of the successPolicy compared with `rules`.
+But, we decided to adopt the API name, `rules` to keep consistency with the existing `podFailurePolicy.rules`.
+
+```golang
+// SuccessPolicy describes when a Job can be declared as succeeded based on the success of some indexes.
+type SuccessPolicy struct {
+	// Criteria represents the list of alternative criteria for declaring the jobs 
+	// as successful before its completion. Once any of the criteria are met,
+	// the "SuccessCriteriaMet" condition is added, and the lingering pods are removed.
+	// The terminal state for such a job has the "Complete" condition.
+	// Additionally, these criteria are evaluated in order; Once the job meets one of the criteria,
+	// other criteria are ignored.
+	//
+	// +optional
+	Criteria []SuccessPolicyCriteria
+}
+
+...
+
+// These are valid conditions of a job.
+const (
+	// JobSuccessCriteriaMet means the job has been succeeded.
+	JobSucceessCriteriaMet JobConditionType = "SuccessCriteriaMet"
+	...
+)
+```
+
 ### Hold succeededIndexes as []int typed in successPolicy
 
 We can hold the `succeededIndexes` as []int typed that a job can be declared as succeeded.
 
 ```golang
 // SuccessPolicyOnSucceededPods describes a Job can be succeeded based on succeeded pods.
-type SuccessPolicyCriterion struct {
+type SuccessPolicyCriteria struct {
```

Comment:
And these? I get they are alternatives, but it would be more readable if we do not combine the alternatives.

Reply:
It makes sense.

Reply:
Done.